### ISSUE TITLE
bug: Exporter ok?

### ISSUE THREAD

**Issue by ile** [uxbox/uxbox#3294]
### Steps To Reproduce

I'm running Penpot so that I run every part separately (backend: ./scripts/start-dev, frontend: npm run watch-main, exporter: ./scripts/wait-and-start.sh), and I'm not really sure, if I'm doing something wrong here.

But, when I do "Export selection" in the UI, I get to the point where the backend is doing something and then gives this error/stack trace:

Note that I added some `(l/info :hint "get-file-permissions" :profile-id profile-id :file-id file-id)` calls there:

```
[2023-06-13 04:31:20.679] I app.rpc.commands.files - hint="get-permissions", profile-id=#uuid "8e6616bc-7fbf-8044-8002-5d759697a38a", file-id="caabb818-e7a6-818c-8002-88eaf6ff60f0"
[2023-06-13 04:31:20.679] I app.rpc.commands.files - hint="get-file-permissions", profile-id=#uuid "8e6616bc-7fbf-8044-8002-5d759697a38a", file-id="caabb818-e7a6-818c-8002-88eaf6ff60f0"
[2023-06-13 04:31:20.683] I app.rpc.commands.files - hint="get-permissions", profile-id=#uuid "8e6616bc-7fbf-8044-8002-5d759697a38a", file-id=#uuid "caabb818-e7a6-818c-8002-88eaf6ff60f0"
[2023-06-13 04:31:20.683] I app.rpc.commands.files - hint="get-file-permissions", profile-id=#uuid "8e6616bc-7fbf-8044-8002-5d759697a38a", file-id=#uuid "caabb818-e7a6-818c-8002-88eaf6ff60f0"
[2023-06-13 04:31:20.688] E app.http.errors - hint="PSQL error", message="ERROR: operator does not exist: uuid = character varying\n  Hint: No operator matches the given name and argument types. You might need to add explicit type casts.\n  Position: 125"
SUMMARY:
 →  org.postgresql.util.PSQLException: ERROR: operator does not exist: uuid = character varying (QueryExecutorImpl.java:2713)
DETAIL:
 →  org.postgresql.util.PSQLException: ERROR: operator does not exist: uuid = character varying (QueryExecutorImpl.java:2713)
    at: org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2713)
        org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2401)
        org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:368)
        org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:498)
        org.postgresql.jdbc.PgStatement.execute(PgStatement.java:415)
        org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:190)
        org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:177)
        com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
        com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java:-1)
        next.jdbc.result_set$stmt__GT_result_set.invokeStatic(result_set.clj:669)
        next.jdbc.result_set$stmt__GT_result_set.invoke(result_set.clj:664)
        next.jdbc.result_set$eval16183$fn__16191.invoke(result_set.clj:897)
        next.jdbc.protocols$eval15468$fn__15469$G__15459__15478.invoke(protocols.clj:34)
        next.jdbc$execute_BANG_.invokeStatic(jdbc.clj:253)
        next.jdbc$execute_BANG_.invoke(jdbc.clj:237)
        app.db$exec_BANG_.invokeStatic(db.clj:232)
        app.db$exec_BANG_.invoke(db.clj:230)
        app.rpc.commands.files$get_file_permissions.invokeStatic(files.clj:114)
        app.rpc.commands.files$get_file_permissions.invoke(files.clj:108)
        app.rpc.commands.files$get_permissions.invokeStatic(files.clj:122)
        app.rpc.commands.files$get_permissions.invoke(files.clj:119)
        clojure.lang.AFn.applyToHelper(AFn.java:160)
        clojure.lang.AFn.applyTo(AFn.java:144)
        clojure.core$apply.invokeStatic(core.clj:669)
        clojure.core$apply.invoke(core.clj:662)
        app.rpc.permissions$make_read_predicate_fn$check__34438.doInvoke(permissions.clj:73)
        clojure.lang.RestFn.applyTo(RestFn.java:139)
        clojure.core$apply.invokeStatic(core.clj:667)
        clojure.core$apply.invoke(core.clj:662)
        app.rpc.permissions$make_check_fn$fn__34447.doInvoke(permissions.clj:90)
        clojure.lang.RestFn.invoke(RestFn.java:436)
        app.rpc.commands.files$sm$get_page.invokeStatic(files.clj:523)
        app.rpc.commands.files$sm$get_page.invoke(files.clj:509)
        app.rpc$wrap_metrics$fn__45343.invoke(rpc.clj:99)
        app.rpc$wrap_audit$fn__45350.invoke(rpc.clj:122)
        app.rpc$wrap_params_validation$fn__45360.invoke(rpc.clj:151)
        app.rpc$wrap_authentication$fn__45346.invoke(rpc.clj:114)
        clojure.core$partial$fn__5908.invoke(core.clj:2641)
        app.rpc$rpc_handler.invokeStatic(rpc.clj:89)
        app.rpc$rpc_handler.invoke(rpc.clj:70)
        clojure.core$partial$fn__5908.invoke(core.clj:2641)
        app.http.session$wrap_authz$fn__22978.invoke(session.clj:241)
        clojure.core$partial$fn__5908.invoke(core.clj:2640)
        promesa.util.Supplier.get(util.cljc:34)
        java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
        java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        java.lang.Thread.run(Thread.java:1589)
```

Changes I made (in `backend/src/app/rpc/commands/files.clj`):

```clojure
(defn get-file-permissions
  [conn profile-id file-id]
  (l/info :hint "get-file-permissions" :profile-id profile-id :file-id file-id)
  (when (and profile-id file-id)
    (db/exec! conn [sql:file-permissions
                    file-id profile-id
                    file-id profile-id
                    file-id profile-id])))
```

Feel free to close, if my configuration is wrong here (hint appreciated what might be wrong). 

Also I'm not sure why this function is called twice and the file-id seems to differ a bit between the calls.

Thanks.

**Comment by superalex** [comment]
Hmmmmm, 

That's strange @ile 

I assume you are using the develop branch, isn't it?, also, could you please check what postgresql version you are using?

Did you try to run directly the development mode using "./manage.sh run-devenv"?

**Comment by ile** [comment]
Hello @superalex!

Yes, develop branch. I also sync with the penpot repo (remote) every time I do anything.

I have postgresql 15.3. Initially I had a problem with the dev-mode (probably my fault), but I'll try it again.

I see this in the postgres window:

```
2023-06-17 08:39:05.704 EEST [55812] ERROR:  operator does not exist: uuid = character varying at character 125
2023-06-17 08:39:05.704 EEST [55812] HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
2023-06-17 08:39:05.704 EEST [55812] STATEMENT:  select fpr.is_owner,
	          fpr.is_admin,
	          fpr.can_edit
	     from file_profile_rel as fpr
	    where fpr.file_id = $1
	      and fpr.profile_id = $2
	   union all
	   select tpr.is_owner,
	          tpr.is_admin,
	          tpr.can_edit
	     from team_profile_rel as tpr
	    inner join project as p on (p.team_id = tpr.team_id)
	    inner join file as f on (p.id = f.project_id)
	    where f.id = $3
	      and tpr.profile_id = $4
	   union all
	   select ppr.is_owner,
	          ppr.is_admin,
	          ppr.can_edit
	     from project_profile_rel as ppr
	    inner join file as f on (f.project_id = ppr.project_id)
	    where f.id = $5
	      and ppr.profile_id = $6
```

I'll try with the proper dev-mode...

**Comment by ile** [comment]
I did a fresh `git clone` of the penpot repo (develop branch). 

I'm running `./manage.sh run-devenv` with it, but the exporter doesn't seem to work:

![image](https://github.com/penpot/penpot/assets/433707/08940e5b-79da-4b83-b844-029c0cd8aa5f)

I can't see anything extra in the backend window. But I see this in the exporter window:

```

DBG [app.handlers] hint="process-request", cmd=:export-shapes
INF [app.browser] origin="factory", action="create", browser-id=5
INF [app.browser] origin="factory", action="validate", browser-id=5, obj=#object[Browser [object Object]]
TRC [app.browser] hint="exec:handle:start", browser-id=5
INF [app.renderer.svg] uri=#lambdaisland/uri "http://localhost:3449/render.html?file-id=f340596f-7c67-810c-8002-ad485486a38a&page-id=f340596f-7c67-810c-8002-ad485486d2f8&render-embed=true&object-id=9c9e6218-e42c-805b-8002-ad4857730f6e&route=objects"
TRC [app.util.shell] hint="schedule tempfile deletion", path="/tmp/penpot/penpot.tmp.render.svg.3adaf009-a071-806b-8002-ad4b123373d6-0.svg", scheduled-at="Sat Jun 17 2023 07:47:31 GMT+0000 (Coordinated Universal Time)"
INF [app.browser] origin="factory", action="destroy", browser-id=5
ERR [app.handlers.export-shapes] hint="unexpected error on export multiple"
  locator.locator: Timeout 10000ms exceeded.
  =========================== logs ===========================
  waiting for locator('#screenshot-9c9e6218-e42c-805b-8002-ad4857730f6e') to be visible
  ============================================================
      at Function.app.browser.wait_for [as cljs$core$IFn$_invoke$arity$2] (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/browser.cljs:53:18)
      at Function.cljs$core$IFn$_invoke$arity$1 (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/browser.cljs:51:15)
      at f (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/renderer/svg.cljs:311:15)
      at Object.resolve (/home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl.cljc:105:29)
      at f (/home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl/promise.js:278:25)
      at /home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl/promise.js:179:35
ERR [app.handlers] hint="unexpected internal error"
  locator.locator: Timeout 10000ms exceeded.
  =========================== logs ===========================
  waiting for locator('#screenshot-9c9e6218-e42c-805b-8002-ad4857730f6e') to be visible
  ============================================================
      at Function.app.browser.wait_for [as cljs$core$IFn$_invoke$arity$2] (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/browser.cljs:53:18)
      at Function.cljs$core$IFn$_invoke$arity$1 (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/browser.cljs:51:15)
      at f (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/renderer/svg.cljs:311:15)
      at Object.resolve (/home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl.cljc:105:29)
      at f (/home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl/promise.js:278:25)
      at /home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl/promise.js:179:35
```

I seem to get this same error with my own copy of the source code too. So I don't see that postgres error now. Maybe it doesn't get that far.

**Comment by superalex** [comment]
I'm glad the postgres error isn't appearing anymore 

Could you please check accessing directly to http://localhost:3449/render.html?file-id=f340596f-7c67-810c-8002-ad485486a38a&page-id=f340596f-7c67-810c-8002-ad485486d2f8&render-embed=true&object-id=9c9e6218-e42c-805b-8002-ad4857730f6e&route=objects ? (that's the url exporter is trying to render)

**Comment by ile** [comment]
Now, what happens:

- With the penpot repo (latest develop branch) and `./manage run-devenv` ... export seems to work. SVG file is downloaded.
- With my own copy of the source (latest penpot develop branch synced) + some irrelevant changes and `./manage run-devenv` ... export still gives this (in export window):

```
INF [app.browser] origin="factory", action="destroy", browser-id=4
ERR [app.handlers.export-shapes] hint="unexpected error on export multiple"
  page.page: Timeout 20000ms exceeded.
  =========================== logs ===========================
  navigating to "https://ile.aamu.app/designs/penpot/render.html?file-id=e0bf020d-ee43-8188-8002-ad5681fd21b8&page-id=e0bf020d-ee43-8188-8002-ad5681fd21b9&render-embed=true&object-id=99766736-c947-809c-8002-ad5684975666&route=objects", waiting until "networkidle"
  ============================================================
      at Function.app.browser.nav_BANG_ [as cljs$core$IFn$_invoke$arity$3] (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/browser.cljs:44:15)
      at Function.cljs$core$IFn$_invoke$arity$2 (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/browser.cljs:42:16)
      at f (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/renderer/svg.cljs:335:15)
      at Object.resolve (/home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl.cljc:105:29)
      at f (/home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl/promise.js:278:25)
      at /home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl/promise.js:179:35
ERR [app.handlers] hint="unexpected internal error"
  page.page: Timeout 20000ms exceeded.
  =========================== logs ===========================
  navigating to "https://ile.aamu.app/designs/penpot/render.html?file-id=e0bf020d-ee43-8188-8002-ad5681fd21b8&page-id=e0bf020d-ee43-8188-8002-ad5681fd21b9&render-embed=true&object-id=99766736-c947-809c-8002-ad5684975666&route=objects", waiting until "networkidle"
  ============================================================
      at Function.app.browser.nav_BANG_ [as cljs$core$IFn$_invoke$arity$3] (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/browser.cljs:44:15)
      at Function.cljs$core$IFn$_invoke$arity$2 (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/browser.cljs:42:16)
      at f (/home/penpot/penpot/exporter/target/app/cljs-runtime/app/renderer/svg.cljs:335:15)
      at Object.resolve (/home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl.cljc:105:29)
      at f (/home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl/promise.js:278:25)
      at /home/penpot/penpot/exporter/target/app/cljs-runtime/promesa/impl/promise.js:179:35
```

If I go to this url https://ile.aamu.app/designs/penpot/render.html?file-id=e0bf020d-ee43-8188-8002-ad5681fd21b8&page-id=e0bf020d-ee43-8188-8002-ad5681fd21b9&render-embed=true&object-id=99766736-c947-809c-8002-ad5684975666&route=objects with a browser, I see the svg rendered correctly in the browser window.

So ... still something,

**Comment by superalex** [comment]
That url https://ile.aamu.app/designs/penpot/render.html must be visible from the exporter, when you use the first option ("With the penpot repo (latest develop branch) and ./manage run-devenv") are you using that url too?

**Comment by ile** [comment]
I'm not 100% sure how exporter sees the url. But I see the url fine on my local machine - the DNS, hostname etc. are pointing to rght places and the pathname works too, in the browser.

I can put some debugging info there and see if I can track the error.

Note that this same error happened earlier (it didn't happen with the latest code) with the fresh penpot too, when I was using the url `http://localhost:3449`. It's strange that it still happens on my code, when it apparently got fixed (?) in penpot code.

**Comment by ile** [comment]
It's possible that the problem is in the domain name, which is local (127.0.0.1). 

We can put this on hold until I get to test on the production server with real setup.

**Comment by superalex** [comment]
Sure...Any way, ./manage.sh run-devenv commands are designed just to run penpot in develop-mode, for the production server you should use something more robust.

**Comment by ile** [comment]
I will, I have built the docker images with ./manage build-bundle and ./manage build-docker-images ... and then run them with the docker/images/docker-compose.yaml.

The tricky part is the domain name, which should support wildcard domains.

**Comment by superalex** [comment]
ok!

**Comment by ile** [comment]
Hi @superalex,

I've now deployed Penpot to the production server (https://aamu.app - you are free to try it if you want to sign up), and still having problems with the exporter. Seems like there is an API call, which should go to backend, but it goes to file system directly. Line from docker compose logs:

`images-penpot-frontend-1  | 2023/07/03 01:24:09 [error] 9#9: *193 open() "/var/www/app/designs/penpot/api/rpc/command/get-page" failed (2: No such file or directory),`

Thumbnails work fine in the dashboard. Also downloading files (.penpot and .zip) from the dashboard seem to work fine. It's just the Export in the editing UI that doesn't seem to work.

Any help would be very much appreciated.

![image](https://github.com/penpot/penpot/assets/433707/36e02072-efa3-49db-8f3f-940f14fa190c)


docker compose logs:

```
images-penpot-exporter-1  | DBG [app.handlers] hint="process-request", cmd=:export-shapes
images-penpot-frontend-1  | 172.19.0.1 - - [03/Jul/2023:01:24:08 +0000] "POST /api/export HTTP/1.1" 200 125 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
images-penpot-exporter-1  | INF [app.renderer.svg] urix=#lambdaisland/uri "http://penpot-frontend/designs/penpot/render.html?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa&page-id=8e46b006-4b96-801c-8002-bd0ee09a3907&render-embed=true&object-id=51819144-a7be-8000-8002-c190789cf2aa&object-id=a60cabcb-ad34-8050-8002-bd0eef3e550e&object-id=a60cabcb-ad34-8050-8002-bd0ee502dc1b&route=objects"
images-penpot-exporter-1  | INF [app.renderer.svg] public-uri=#lambdaisland/uri "http://penpot-frontend"
images-penpot-exporter-1  | INF [app.browser] origin="factory", action="create", browser-id=4
images-penpot-exporter-1  | INF [app.browser] origin="factory", action="validate", browser-id=4, obj=#object[Browser [object Object]]
images-penpot-exporter-1  | TRC [app.browser] hint="exec:handle:start", browser-id=4
images-penpot-exporter-1  | INF [app.renderer.svg] uri=#lambdaisland/uri "http://penpot-frontend/designs/penpot/render.html?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa&page-id=8e46b006-4b96-801c-8002-bd0ee09a3907&render-embed=true&object-id=51819144-a7be-8000-8002-c190789cf2aa&object-id=a60cabcb-ad34-8050-8002-bd0eef3e550e&object-id=a60cabcb-ad34-8050-8002-bd0ee502dc1b&route=objects"
images-penpot-frontend-1  | 172.19.0.2 - - [03/Jul/2023:01:24:08 +0000] "GET /designs/penpot/render.html?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa&page-id=8e46b006-4b96-801c-8002-bd0ee09a3907&render-embed=true&object-id=51819144-a7be-8000-8002-c190789cf2aa&object-id=a60cabcb-ad34-8050-8002-bd0eef3e550e&object-id=a60cabcb-ad34-8050-8002-bd0ee502dc1b&route=objects HTTP/1.1" 200 336 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.3729.169 Safari/537.36"
images-penpot-frontend-1  | 172.19.0.2 - - [03/Jul/2023:01:24:08 +0000] "GET /designs/penpot/js/config.js?ts=1688032192061 HTTP/1.1" 200 121 "http://penpot-frontend/designs/penpot/render.html?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa&page-id=8e46b006-4b96-801c-8002-bd0ee09a3907&render-embed=true&object-id=51819144-a7be-8000-8002-c190789cf2aa&object-id=a60cabcb-ad34-8050-8002-bd0eef3e550e&object-id=a60cabcb-ad34-8050-8002-bd0ee502dc1b&route=objects" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.3729.169 Safari/537.36"
images-penpot-frontend-1  | 172.19.0.2 - - [03/Jul/2023:01:24:08 +0000] "GET /designs/penpot/js/shared.698bf49c7.js HTTP/1.1" 200 861017 "http://penpot-frontend/designs/penpot/render.html?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa&page-id=8e46b006-4b96-801c-8002-bd0ee09a3907&render-embed=true&object-id=51819144-a7be-8000-8002-c190789cf2aa&object-id=a60cabcb-ad34-8050-8002-bd0eef3e550e&object-id=a60cabcb-ad34-8050-8002-bd0ee502dc1b&route=objects" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.3729.169 Safari/537.36"
images-penpot-frontend-1  | 172.19.0.2 - - [03/Jul/2023:01:24:09 +0000] "GET /designs/penpot/js/polyfills.js?ts=1688032192061 HTTP/1.1" 200 1487 "http://penpot-frontend/designs/penpot/render.html?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa&page-id=8e46b006-4b96-801c-8002-bd0ee09a3907&render-embed=true&object-id=51819144-a7be-8000-8002-c190789cf2aa&object-id=a60cabcb-ad34-8050-8002-bd0eef3e550e&object-id=a60cabcb-ad34-8050-8002-bd0ee502dc1b&route=objects" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.3729.169 Safari/537.36"
images-penpot-frontend-1  | 172.19.0.2 - - [03/Jul/2023:01:24:09 +0000] "GET /designs/penpot/js/render.698bf49c7.js HTTP/1.1" 200 15682 "http://penpot-frontend/designs/penpot/render.html?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa&page-id=8e46b006-4b96-801c-8002-bd0ee09a3907&render-embed=true&object-id=51819144-a7be-8000-8002-c190789cf2aa&object-id=a60cabcb-ad34-8050-8002-bd0eef3e550e&object-id=a60cabcb-ad34-8050-8002-bd0ee502dc1b&route=objects" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.3729.169 Safari/537.36"
images-penpot-frontend-1  | 2023/07/03 01:24:09 [error] 9#9: *195 open() "/var/www/app/designs/penpot/api/rpc/command/get-font-variants" failed (2: No such file or directory), client: 172.19.0.2, server: _, request: "GET /designs/penpot/api/rpc/command/get-font-variants?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa HTTP/1.1", host: "penpot-frontend"
images-penpot-frontend-1  | 172.19.0.2 - - [03/Jul/2023:01:24:09 +0000] "GET /designs/penpot/api/rpc/command/get-font-variants?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa HTTP/1.1" 404 178 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.3729.169 Safari/537.36"
images-penpot-frontend-1  | 2023/07/03 01:24:09 [error] 9#9: *193 open() "/var/www/app/designs/penpot/api/rpc/command/get-page" failed (2: No such file or directory), client: 172.19.0.2, server: _, request: "GET /designs/penpot/api/rpc/command/get-page?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa&page-id=8e46b006-4b96-801c-8002-bd0ee09a3907 HTTP/1.1", host: "penpot-frontend"
images-penpot-frontend-1  | 172.19.0.2 - - [03/Jul/2023:01:24:09 +0000] "GET /designs/penpot/api/rpc/command/get-page?file-id=8e46b006-4b96-801c-8002-bd0ee0979efa&page-id=8e46b006-4b96-801c-8002-bd0ee09a3907 HTTP/1.1" 404 178 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.3729.169 Safari/537.36"
images-penpot-exporter-1  | TRC [app.util.shell] hint="schedule tempfile deletion", path="/tmp/penpot/penpot.tmp.render.svg.9db0e0dd-bdd7-8010-8002-c19a804871d1-0.svg", scheduled-at="Mon Jul 03 2023 02:24:10 GMT+0000 (Coordinated Universal Time)"
images-penpot-exporter-1  | INF [app.browser] origin="factory", action="destroy", browser-id=4
images-penpot-exporter-1  | ERR [app.handlers.export-shapes] hint="unexpected error on multiple exportation"
images-penpot-exporter-1  |   locator.waitFor: Timeout 10000ms exceeded.
images-penpot-exporter-1  |   =========================== logs ===========================
images-penpot-exporter-1  |   waiting for locator('#screenshot-51819144-a7be-8000-8002-c190789cf2aa') to be visible
images-penpot-exporter-1  |     locator resolved to hidden <svg fill="none" width="0.01" version="1.1" height="0…></svg>
images-penpot-exporter-1  |   ============================================================
images-penpot-exporter-1  |       at app.browser.wait_for.cljs$core$IFn$_invoke$arity$2 (/opt/penpot/exporter/app.js:69532:21)
images-penpot-exporter-1  |       at app.browser.wait_for.cljs$core$IFn$_invoke$arity$1 (/opt/penpot/exporter/app.js:69526:31)
images-penpot-exporter-1  |       at /opt/penpot/exporter/app.js:74147:90
images-penpot-exporter-1  |       at Object.resolve (/opt/penpot/exporter/app.js:68534:107)
images-penpot-exporter-1  |       at $jscomp$scope$84173391$24$processNextTick (/opt/penpot/exporter/app.js:67609:78)
images-penpot-exporter-1  |       at /opt/penpot/exporter/app.js:67706:36
```

The request in the browser (as "Copy as fetch"). 

```
fetch("https://ilkkah.aamu.app/designs/penpot/api/export", {
  "headers": {
    "accept": "*/*",
    "accept-language": "fi,en-US;q=0.9,en;q=0.8",
    "cache-control": "no-cache",
    "content-type": "application/transit+json",
    "pragma": "no-cache",
    "sec-ch-ua": "\"Not.A/Brand\";v=\"8\", \"Chromium\";v=\"114\", \"Brave\";v=\"114\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Linux\"",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "same-origin",
    "sec-gpc": "1",
    "x-frontend-version": "1.20.0-10395-g698bf49c7"
  },
  "referrerPolicy": "no-referrer",
  "body": "{\"~:wait\":false,\"~:exports\":[{\"~:suffix\":\"\",\"~:file-id\":\"~u8e46b006-4b96-801c-8002-bd0ee0979efa\",\"~:object-id\":\"~u51819144-a7be-8000-8002-c190789cf2aa\",\"~:scale\":1,\"~:name\":\"Ellipse\",\"~:type\":\"~:svg\",\"~:page-id\":\"~u8e46b006-4b96-801c-8002-bd0ee09a3907\"},{\"~:suffix\":\"\",\"~:file-id\":\"~u8e46b006-4b96-801c-8002-bd0ee0979efa\",\"~:object-id\":\"~ua60cabcb-ad34-8050-8002-bd0eef3e550e\",\"~:scale\":1,\"~:name\":\"Rectangle\",\"~:type\":\"~:svg\",\"~:page-id\":\"~u8e46b006-4b96-801c-8002-bd0ee09a3907\"},{\"~:suffix\":\"\",\"~:file-id\":\"~u8e46b006-4b96-801c-8002-bd0ee0979efa\",\"~:object-id\":\"~ua60cabcb-ad34-8050-8002-bd0ee502dc1b\",\"~:scale\":1,\"~:name\":\"Ellipse\",\"~:type\":\"~:svg\",\"~:page-id\":\"~u8e46b006-4b96-801c-8002-bd0ee09a3907\"}],\"~:cmd\":\"~:export-shapes\",\"~:profile-id\":\"~u8e46b006-4b96-801c-8002-bd06ebd170ed\"}",
  "method": "POST",
  "mode": "cors",
  "credentials": "include"
});
```

Request headers:

```
POST /designs/penpot/api/export HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: fi,en-US;q=0.9,en;q=0.8
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 789
Cookie: sessionid=s%3Azrw4TyJRWuUJjJ2Di_x_Ey7mLtGxK5qe.E%2FwWXvBW%2FJ%2FiVSK%2Bp3L8QX70ytRkVKFzvHFYhaNunKY; cids=j%3A%5B%22c78de4b4-c1ed-41e5-83af-a27352c12c31%22%5D; auth-token=eyJhbGciOiJBMjU2S1ciLCJlbmMiOiJBMjU2R0NNIn0.XusDFop3Yak2q_iJ_GEgu2gIqCLj2VTjMYZdxd7OxuadedPYBC2ZzQ.lvIB0x0YLzM-RgP1.e9RBabolFGBn4zygzOwYDzBcseCxX6b5ljNelJYmEodbIFap0QdvkXKv_GlUbh6v5YXz0BY2UAeNcfJsniDDq3aiN0KkvpvYHQuh54sS1EDGqklb5g1FI0JkttHq_NWnNsqcDxWMHL97Pw.fEAHsztlLaOymD-QGTqgRQ
Host: ilkkah.aamu.app
Origin: https://ilkkah.aamu.app
Pragma: no-cache
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-origin
Sec-GPC: 1
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
content-type: application/transit+json
sec-ch-ua: "Not.A/Brand";v="8", "Chromium";v="114", "Brave";v="114"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "Linux"
x-frontend-version: 1.20.0-10395-g698bf49c7
```

Response headers:

```
HTTP/1.1 200 OK
Server: nginx/1.21.4
Date: Mon, 03 Jul 2023 01:09:16 GMT
Content-Type: application/transit+json
Transfer-Encoding: chunked
Connection: keep-alive
Vary: Accept-Encoding
Content-Encoding: gzip
```

Relevant parts in nginx.conf:

```
upstream penpot-frontend {
        server localhost:3449;
}


server {

	... stuff ...

	# Penpot
	charset utf-8;

	proxy_http_version 1.1;
	proxy_set_header Host $http_host;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Scheme $scheme;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

	etag off;

	location @handle_redirect {
		set $redirect_uri "$upstream_http_location";
		set $redirect_host "$upstream_http_x_host";
		set $redirect_cache_control "$upstream_http_cache_control";

		proxy_buffering off;

		proxy_set_header Host "$redirect_host";
		proxy_hide_header etag;
		proxy_hide_header x-amz-id-2;
		proxy_hide_header x-amz-request-id;
		proxy_hide_header x-amz-meta-server-side-encryption;
		proxy_hide_header x-amz-server-side-encryption;
		proxy_pass $redirect_uri;

		add_header x-internal-redirect "$redirect_uri";
		add_header x-cache-control "$redirect_cache_control";
		add_header cache-control "$redirect_cache_control";
	}

	location /designs/penpot/assets {
		proxy_pass http://penpot-frontend/assets;
		recursive_error_pages on;
		proxy_intercept_errors on;
		error_page 301 302 307 = @handle_redirect;
	}

	location /internal/assets {
		internal;
		alias /opt/data/assets;
		add_header x-internal-redirect "$upstream_http_x_accel_redirect";
	}

	location /designs/penpot/api/export {
		proxy_pass http://penpot-frontend/api/export;
	}

	location /designs/penpot/api {
		proxy_pass http://penpot-frontend/api;
	}

	location /designs/penpot/ws/notifications {
		proxy_set_header Upgrade $http_upgrade;
		proxy_set_header Connection 'upgrade';
		proxy_pass http://penpot-frontend/ws/notifications;
	}

	location /designs/penpot/ {
		proxy_pass http://penpot-frontend;
	}

	... stuff ...
}
